lib/GenomeFileUtil/core/GenomeInterface.py

```diff
@@ -128,54 +171,70 @@ def get_one_genome(self, params):
-        logging.info('start saving genome object')
-        self._validate_save_one_genome_params(params)
-        workspace = params['workspace']
-        name = params['name']
-        data = params['data']
-        # XXX there is no `workspace_datatype` param in the spec
-        ws_datatype = params.get('workspace_datatype', "KBaseGenomes.Genome")
-        # XXX there is no `meta` param in the spec
-        meta = params.get('meta', {})
-        if "AnnotatedMetagenomeAssembly" in ws_datatype:
-            if params.get('upgrade') or 'feature_counts' not in data:
-                data = self._update_metagenome(data)
-        else:
-            if params.get('upgrade') or 'feature_counts' not in data:
-                data = self._update_genome(data)
-
-        # check all handles point to shock nodes owned by calling user
-        self._own_handle(data, 'genbank_handle_ref')
-        self._own_handle(data, 'gff_handle_ref')
-        if "AnnotatedMetagenomeAssembly" not in ws_datatype:
-            self._check_dna_sequence_in_features(data)
-            data['warnings'] = self.validate_genome(data)
-
-        # sort data
-        data = GenomeUtils.sort_dict(data)
-        # dump genome to scratch for upload
-        data_path = os.path.join(self.scratch, name + ".json")
-        json.dump(data, open(data_path, 'w'))
-        if 'hidden' in params and str(params['hidden']).lower() in ('yes', 'true', 't', '1'):
-            hidden = 1
-        else:
-            hidden = 0
+    def _save_genome_mass(self, params, validate_genome=True):
 
-        if isinstance(workspace, int) or workspace.isdigit():
-            workspace_id = workspace
-        else:
-            workspace_id = self.dfu.ws_name_to_id(workspace)
-
-        save_params = {'id': workspace_id,
-                       'objects': [{'type': ws_datatype,
-                                    'data_json_file': data_path,
-                                    'name': name,
-                                    'meta': meta,
-                                    'hidden': hidden}]}
-        dfu_oi = self.ws_large_data.save_objects(save_params)[0]
-        returnVal = {'info': dfu_oi, 'warnings': data.get('warnings', [])}
-        return returnVal
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        objects = []
+        warnings = []
+
+        for input_params in inputs:
+
+            obj = {}
+
+            # retrive required params
+            name = input_params['name']
+            data = dict(input_params['data'])
+
+            # XXX there is no `workspace_datatype` param in the spec
+            # NOTE: This allows a user to specify any arbitrary workspace type which could cause,
+            # in the worst case, data corruption. It should be removed from the API
+            # (note it is not currently documented) so users cannot access it.
+            ws_datatype = input_params.get('workspace_datatype', "KBaseGenomes.Genome")
```

Comment:
Just to make sure we're not doing something backwards incompatible - was, and is, this always going to be KBG.Genome in practice? E.g. is, or was, there anyway to specify some argument in the external API and change this value?

If it's always a regular Genome, we should get rid of the if statements around its contents below

Reply:
https://github.com/kbaseapps/GenomeFileUtil/pull/212#discussion_r2216647797

Reply:
:+1: